problem_builder/tests/unit/test_instructor_tool.py

```diff
@@ -0,0 +1,226 @@
+
+        # Check size of block_tree
+        num_blocks = len(self.block.children) + 1
+        self.assertEqual(len(block_tree), num_blocks)
+
+        # Check block_tree for root entry
+        # - Does block_tree contain single entry whose `id` matches usage_id of instructor block?
+        matching_blocks = [block for block in block_tree if block['id'] == self.block.scope_ids.usage_id]
+        self.assertEqual(len(matching_blocks), 1)
+        # # - Is `name` of that entry set to "All"?
+        matching_block = matching_blocks.pop()
+        self.assertEqual(matching_block['name'], 'All')
+
+        # Check block_tree for children
+        for child in self.block.children:
+            usage_id = child.get('usage_id')
+            attrs = child.get('attrs', {})
+            # - Does block_tree contain single entry whose `id` matches usage_id of current child?
+            matching_blocks = [block for block in block_tree if block['id'] == usage_id]
+            self.assertEqual(len(matching_blocks), 1)
+            # - Is `name` of that entry set to value of current child's preferred_attr?
+            matching_block = matching_blocks.pop()
+            if not attrs:
+                self.assertEqual(matching_block['name'], usage_id)
+            else:
+                preferred_attr = child.get('preferred_attr')
+                self.assertEqual(matching_block['name'], attrs[preferred_attr])
+
+    def test_student_view_block_tree_choice_blocks_excluded(self):
+        """
+        Check if `student_view` method excludes 'pb-choice' blocks when
+        building block tree for render context.
+        """
+        # Pretend that all blocks in self.block.children are of type 'pb-choice:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = 'pb-choice'
+
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree: Should only include root block
+        self.assertEqual(len(block_tree), 1)
+
+    def test_student_view_block_tree_eligible(self):
+        """
+        Check if `student_view` method correctly marks MCQ, Rating, and
+        Answer blocks as eligible when building tree for render
+        context.
+
+        A block is eligible if its type is one of {'pb-mcq', 'pb-rating', 'pb-answer'}.
+        """
+        # Pretend that all blocks in self.block.children are eligible:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = random.choice(['pb-mcq', 'pb-rating', 'pb-answer'])
+
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
```

Comment:
@itsjeyd I'm afraid I don't like the overall approach here - you're essentially testing `build_tree` function that is defined as local in `student_view`. As a result, you've got to patch all the other stuff and call `student_view`. It might make sense to make `build_tree` a method on a block than test it explicitly, without `studen_view` and intercepting template rendering call arguments.

If you do want to test that correct template arguments are passed, it would be simpler than to patch `build_tree` to return whatever you want, and than intercept rendering call.